extras/gizmo/gizmo.js

```diff
@@ -0,0 +1,325 @@
+    /**
+     * The application instance containing the gizmo.
+     *
+     * @type {import('playcanvas').AppBase}
+     */
+    app;
+
+    /**
+     * The camera entity that displays the gizmo.
+     *
+     * @type {Entity}
+     */
+    camera;
+
+    /**
+     * The graph nodes attached to the gizmo.
+     *
+     * @type {import('playcanvas').GraphNode}
+     */
+    nodes = [];
+
+    /**
+     * The root gizmo entity.
+     *
+     * @type {import('playcanvas').Entity}
+     */
+    gizmo;
+
+    /**
+     * @typedef IntersectData
+     * @property {import('./mesh-tri-data.js').MeshTriData[]} meshTriDataList -
+     * The array of {@link MeshTriData}
+     * @property {import('playcanvas').GraphNode} parent - The mesh parent node.
+     * @property {import('playcanvas').MeshInstance[]} meshInstances -
+     * array of mesh instances for rendering
+     */
+    /**
+     * The intersection data object.
+     *
+     * @type {IntersectData[]}
+     */
+    intersectData = [];
+
+    /**
+     * Creates a new Gizmo object.
+     *
+     * @param {import('playcanvas').AppBase} app - The application instance.
+     * @param {import('playcanvas').CameraComponent} camera - The camera component.
+     * @example
+     * const gizmo = new pcx.Gizmo(app, camera);
+     */
+    constructor(app, camera) {
+        super();
+
+        this.app = app;
+        this.camera = camera;
+
+        this._createLayer();
+        this._createGizmo();
+
+        this._updateScale();
+
+        this._onPointerDown = (e) => {
+            if (!this.gizmo.enabled || document.pointerLockElement) {
+                return;
+            }
+            const selection = this._getSelection(e.clientX, e.clientY);
+            if (selection[0]) {
+                e.preventDefault();
+            }
+            this.fire('pointer:down', e.clientX, e.clientY, selection[0]);
+        };
+        this._onPointerMove = (e) => {
+            if (!this.gizmo.enabled || document.pointerLockElement) {
+                return;
+            }
+            const selection = this._getSelection(e.clientX, e.clientY);
+            if (selection[0]) {
+                e.preventDefault();
+            }
+            this.fire('pointer:move', e.clientX, e.clientY, selection[0]);
+        };
+        this._onPointerUp = (e) => {
+            if (!this.gizmo.enabled || document.pointerLockElement) {
+                return;
+            }
+            this.fire('pointer:up');
+        };
+        this._onKeyDown = (e) => {
+            if (!this.gizmo.enabled) {
+                return;
+            }
+            this.fire('key:down', e.key, e.shiftKey, e.ctrlKey, e.metaKey);
+        };
+        this._onKeyUp = (e) => {
+            if (!this.gizmo.enabled) {
+                return;
+            }
+            this.fire('key:up');
+        };
+
+        app.on('update', () => this._updateScale());
+
+        app.on('destroy', () => this.detach());
+    }
+
+    set coordSpace(value) {
```

Comment:
in public API, we use exported constants instead of strings, otherwise I'm not sure what even the options here are apart from 'world'.